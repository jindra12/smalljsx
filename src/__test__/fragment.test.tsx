// import preview from "jest-preview";
import $ from "jquery";
import * as Small from "../index";
jest.useFakeTimers();

describe("Can use fragments", () => {
    beforeEach(() => {
        document.body.innerHTML = "<div id='root' />";
    });
    afterEach(() => {
        Small.__reset();
    });
    it("Can use simple fragments", () => {
        const Parent: Small.ParentComponent = (props) => {
            return <>{props.children}<div>Not me!</div></>;
        };
        const Component: Small.Component = () => {
            return (
                <Parent>
                    <>Click me!</>
                </Parent>
            );
        };
        Small.mount(<Component />, "#root");
        jest.runAllTimers();
        expect($("#root")[0].outerHTML).toEqual(
            '<div id="root">Click me!<div>Not me!</div></div>'
        );
    });
});
