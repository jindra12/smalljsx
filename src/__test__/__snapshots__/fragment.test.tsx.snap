// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can use fragments Can work with fragments conditionally 1`] = `"<div id="root"><div>00<button id="click">Update</button></div></div>"`;

exports[`Can use fragments Can work with fragments conditionally 2`] = `"<div id="root"><div><p>1</p><div>1</div><button id="click">Update</button></div></div>"`;

exports[`Can use fragments Can work with fragments conditionally 3`] = `"<div id="root"><div>22<button id="click">Update</button></div></div>"`;

exports[`Can use fragments Can work with fragments conditionally 4`] = `"<div id="root"><div><p>3</p><div>3</div><button id="click">Update</button></div></div>"`;
